### ISSUE TITLE
Environment variable can't contain `=`

### ISSUE THREAD

**Issue by Aloso** [nushell/nushell#2023]
**Describe the bug**
I tried to run the following:

```sh
❯ RUSTFLAGS="-C opt-level=0" cargo build
error: Type Error
  ┌─ shell:1:1
  │
1 │ RUSTFLAGS="-C opt-level=0" cargo build
  │ ^^^^^^^^^^^^^^^^^^^^^^^^^^ Expected environment variable assignment, found RUSTFLAGS="-C opt-level=0"
```

**To Reproduce**

Try this command:

```sh
❯ FOO="a=b" ls
```

**Expected behavior**

The env variable `FOO` is set to `a=b`, like in every other shell.

Note that `❯ with-env [FOO "a=b"] { ls }` works, but this three times as long.

**Configuration**
 - OS:
 - Nu version: 0.15.0
 - Optional features: All

**Comment by samhedin** [comment]
The offending line is `let assignment: Vec<_> = lite_pipeline.commands[0].name.split('=').collect();` inside `parse.rs`.
Changing it to `let assignment: Vec<_> = lite_pipeline.commands[0].name.splitn(2, '=').collect();` fixes it, but right before I submitted a PR wit the change I saw that it's very connected to https://github.com/nushell/nushell/issues/1832 so maybe it's better to wait.

**Comment by jonathandturner** [comment]
Still a bug

**Comment by onthebridgetonowhere** [comment]
I cannot reproduce this issue anymore with ver. 0.35. The line mentioned by @samhedin seems to be fixed as per his suggestion. Maybe this can be closed?

**Comment by Aloso** [comment]
> I cannot reproduce this issue anymore with ver. 0.35

Me neither. Let's close this.

**Comment by lily-mara** [comment]
Oh yes, I fixed this in #3869. I didn't realize there was an existing issue
